calculation/sha_calc/sha_calc/directivity/bea20/bea20.py

```diff
@@ -0,0 +1,224 @@
+            type = 1
+        else:
+            type = 2
+
+    # Convert U to S
+    # Limit S to the Smax values for positive and negative numbers
+    Smax1, Smax2 = Smax
+    U, S = np.asarray(U).copy(), np.asarray(U).copy()
+    S[U < Smax1] = Smax1
+    S[U > Smax2] = Smax2
+
+    # Convert U to Ry
+    # Reduce each positive and negative side by absolute value of Smax and remove negative numbers inbetween
+    upos, uneg = U >= 0, U < 0
+    Ry = np.zeros(U.size)
+    Ry[upos] = U[upos] - Smax2
+    Ry[uneg] = abs(U[uneg]) - abs(Smax1)
+    Ry[Ry < 0] = 0
+
+    # Calculate S2
+    if type == 2 and Rake < 0:
+        S = -S
+    Srake = S * math.cos(math.radians(Rake))
+    D = max(D, 3)  # Gets the max value for D, 3 is the minimum
+    S2 = np.sqrt(Srake ** 2 + D ** 2)
+
+    # Predictor variable fs2
+    if type == 1:
+        fs2 = np.log(S2)
+        fsCap = math.log(465)
+        fs2[fs2 > fsCap] = fsCap
+    else:
+        fs2 = np.zeros(U.size)
+        spos, sneg = Srake >= 0, Srake < 0
+        fs2[spos] = np.log(S2[spos])
+        fs2[sneg] = np.log(D)
+        fsCap = math.log(188)
+        fs2[fs2 > fsCap] = fsCap
+
+    # Angular predictor variables
+    T = np.asarray(T).copy()
+    if type == 1:
+        theta = np.nan_to_num(np.arctan(np.divide(T, U)))
+        ftheta = abs(np.cos(2 * theta))
+        fphi = np.ones(U.size)
+    else:
+        tpos, tneg = T > 0, T <= 0
+        phi = np.zeros(U.size)
+
+        phi[tneg] = (
+            np.degrees(np.arctan(np.divide(abs(T[tneg]) + Tbot, Dbot))) + Dip - 90
+        )
+
+        t1 = np.logical_and(tpos, T < Tbot)
+        phi[t1] = 90 - Dip - np.degrees(np.arctan(np.divide(Tbot - T[t1], Dbot)))
+
+        t2 = np.logical_and(tpos, T >= Tbot)
+        phi[t2] = 90 - Dip + np.degrees(np.arctan(np.divide(T[t2] - Tbot, Dbot)))
+
+        phi[phi > 45] = 45
+        fphi = np.cos(2 * phi * np.pi / 180)
+
+        Tmin = 10 * (abs(math.cos(math.radians(Rake))) + 1)
+        T2 = abs(T)
+        T2[T2 < Tmin] = Tmin
+        T2[tpos] = Tmin
+
+        omega = np.arctan(np.divide(T2, Ry))
+        ftheta = np.nan_to_num(np.sin(omega), nan=1)
+
+    # Distance taper
+    R = np.sqrt(T ** 2 + Ry ** 2)  # Distance from surface trace
+    if M < 5:
+        Rmax = 40
+    elif M > 7:
+        Rmax = 80
+    else:
+        Rmax = -60 + 20 * M
+
+    if type == 2:
+        Rmax = Rmax - 20
+
+    AR = -4 * Rmax
+    Footprint = R <= Rmax
+    fdist = np.zeros(R.size)
+    fdist[Footprint] = 1 - np.exp(np.divide(AR, R[Footprint]) - np.divide(AR, Rmax))
+    fG = fs2 * ftheta * fphi
+
+    # Calculate fD
+
+    # Constants
+    Per = np.logspace(-2, 1, 1000)
+    coefb = [-0.0336, 0.5469]  # mag dependence of bmax
+    coefc = [0.2858, -1.2090]  # mag scaling of Tpeak
+    coefd1 = [0.9928, -4.8300]  # mag dependence of fG0 for SOF=1
+    coefd2 = [0.3946, -1.5415]  # mag dependence of fG0 for SOF=2
+    SigG = 0.4653
```

Comment:
Would it be better to keep the constants at the module level? instead of function level.
and make it all upper case except `SigG` to avoid any confusion, maybe something like `SIG_G`?

Reply:
Some of the naming/capitalization in the bea20 function is due to lining up the variables in the Matlab functions and the paper for easier understanding and debugging, was a chat that I had with @claudio525.

Reply:
Yeah I figured after I noticed that Matlab code haha
Matlab code just came after I checked this file lol